internal/discovery/collector/client_test.go

```diff
@@ -24,12 +25,15 @@ func TestCollectorClientTestSuite(t *testing.T) {
 }
 
 func (suite *CollectorClientTestSuite) TestCollectorClientPublishingSuccess() {
-	collectorClient := NewCollectorClient(
-		&Config{
+	httpClient := http.DefaultClient
```

Comment:
Not a biggie, however what about having `httpClient` set in the suite and pass it as the second argument of `NewCollectorClient`?
```go
collectorClient := collector.NewCollectorClient(
    &collector.Config{/** ... */},
    suite.httpClient,
)
```
as in publishing_test.go?

Reply:
No problem! I can change that

Reply:
Done